airbyte_cdk/sources/declarative/retrievers/simple_retriever.py

```diff
@@ -626,6 +627,123 @@ def _to_partition_key(to_serialize: Any) -> str:
+            record = retriever.fetch_one("123", schema)
+
+            record = retriever.fetch_one({"company_id": "123", "property": "status"}, schema)
+
+        Note:
+            This implementation uses convention-based path construction (Option B from design). (important-comment)
+            For simple PKs: appends /{pk_value} to base path (important-comment)
+            For composite PKs: appends /{value1}/{value2}/... in key order (important-comment)
+
+            Alternative approaches that could be implemented in the future: (important-comment)
+            - Path template interpolation: Use a configurable template like "{base_path}/{id}" (important-comment)
+              See: https://github.com/airbytehq/airbyte-python-cdk/issues/833#phase-1a (important-comment)
+            - Field path configuration: Allow specifying which response field contains the record (important-comment)
+              for APIs that wrap single records in envelopes like {"data": {...}} (important-comment)
+        """
+        # Get the base path from the requester
+        base_path = self.requester.get_path(
+            stream_state={},
+            stream_slice=StreamSlice(partition={}, cursor_slice={}),
+            next_page_token=None,
+        )
+
+        if isinstance(pk_value, str):
+            fetch_path = f"{base_path.rstrip('/')}/{str(pk_value).lstrip('/')}"
+        elif isinstance(pk_value, Mapping):
+            sorted_values = [str(pk_value[key]).lstrip("/") for key in sorted(pk_value.keys())]
+            pk_path_segment = "/".join(sorted_values)
+            fetch_path = f"{base_path.rstrip('/')}/{pk_path_segment}"
+        else:
+            raise ValueError(f"pk_value must be a string or dict, got {type(pk_value).__name__}")
+
+        stream_slice = StreamSlice(partition={}, cursor_slice={})
+
+        response = self.requester.send_request(
+            path=fetch_path,
+            stream_state={},
+            stream_slice=stream_slice,
+            next_page_token=None,
+            request_headers=self._request_headers(
+                stream_state={},
+                stream_slice=stream_slice,
+                next_page_token=None,
+            ),
+            request_params=self._request_params(
+                stream_state={},
+                stream_slice=stream_slice,
+                next_page_token=None,
+            ),
+            request_body_data=self._request_body_data(
+                stream_state={},
+                stream_slice=stream_slice,
+                next_page_token=None,
+            ),
+            request_body_json=self._request_body_json(
+                stream_state={},
+                stream_slice=stream_slice,
+                next_page_token=None,
+            ),
+            log_formatter=self.log_formatter,
+        )
+
+        if not response:
+            raise RecordNotFoundException(
+                f"Record with primary key {pk_value} not found (no response)"
+            )
```

Comment:
To confirm: are you sure response is falsey here? (If yes, that's great. I just want to check. And it might be worth a short comment to say so.)